baseplate/integration/pyramid.py

```diff
@@ -116,20 +116,7 @@ def _on_new_request(self, event):
         trace_info = None
         if self.trust_trace_headers:
             try:
-                sampled = request.headers.get("X-Sampled", None)
-                if sampled is not None:
-                    sampled = True if sampled == "1" else False
-                flags = request.headers.get("X-Flags", None)
-                if flags is not None:
-                    flags = int(flags)
-                trace_info = TraceInfo.from_upstream(
-                    trace_id=int(request.headers["X-Trace"]),
-                    parent_id=int(request.headers["X-Parent"]),
-                    span_id=int(request.headers["X-Span"]),
-                    sampled=sampled,
-                    flags=flags,
-                )
-
+                trace_info = TraceInfo.from_http_headers(request.headers)
```

Comment:
It's kinda a bummer to make `TraceInfo` have to be aware of the different protocols we're using; if we added more down the line that'd mean more direct modifications in that class rather than just adding the new protocol definitions. The core stuff is meant to be the idealized form of the tracing and then these integrations are where that meets the actual wire. The centralized extraction logic could potentially be exposed as a helper that these integrations use though?